.github/workflows/pr.yml

```diff
@@ -1,21 +1,30 @@
+---
 name: PR testing
 
-on: [push, pull_request]
+# run on each PR and on merges to main
+on:
+  pull_request: {}
+  push:
+    branches:
+      - main
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     strategy:
       matrix:
-        java: [ '11, '17' ]
+        java:
+          - '11'
+          - '17'
+          - '21'
```

Comment:
Technically Java 21 is a new feature, right? Not sure if the release notes should mention this.

Reply:
This is just testing ezbake itself. While server/db don’t list support for 21 (and very well made need changes due to namespacing changes in 21) this at least gets ezbake ready for it.

Reply:
added it to CHANGELOG.md